source/beamline/BioXAS/BioXASCarbonFilterFarmActuatorFilterControl.cpp

```diff
@@ -76,7 +76,24 @@ void BioXASCarbonFilterFarmActuatorFilterControl::clearFilters()
 
 void BioXASCarbonFilterFarmActuatorFilterControl::setWindowPreference(double filter, int windowIndex)
 {
-	filterWindowPreferenceMap_.insert(filter, windowIndex);
+	if (indices_.contains(windowIndex)) {
+
+		// Identify any other window indices with the given filter value.
+		// Mark these windows as read-only.
+
+		QList<int> filterMatches = indexFilterMap_.keys(filter);
```

Comment:
This is what I meant to do. If there are multiple windows that have the same filter (eg. the upstream actuator on Side has a 50um filter in Top and Bottom windows), you can choose to set a window preference for that particular filter (so the Bottom window is always used when you want a 50um filter). If there is a preference, it should be the only one that is a move destination, the other windows with that filter should be read-only.